{{ cookiecutter.formal_name }}/app/src/main/java/org/beeware/android/MainActivity.java

```diff
@@ -108,27 +113,36 @@ private void setPythonEnvVars(String pythonHome) throws IOException, ErrnoExcept
         Os.setenv("LD_LIBRARY_PATH", this.getApplicationInfo().nativeLibraryDir, true);
         Os.setenv("PYTHONHOME", pythonHome, true);
         Os.setenv("ACTIVITY_CLASS_NAME", "org/beeware/android/MainActivity", true);
+        Log.d(APP_LAUNCH_TAG, "setPythonEnvVars() complete");
     }
 
     private void startPython(Map<String, File> dirs) throws Exception {
+        Log.d(APP_LAUNCH_TAG, "startPython() start");
         this.unpackPython(dirs);
         String pythonHome = dirs.get("stdlib").getAbsolutePath();
         this.setPythonEnvVars(pythonHome);
 
         // `app` is the last item in the sysPath list.
         String sysPath = (pythonHome + "/lib/python3.7/") + ":" + dirs.get("rubicon_java").getAbsolutePath() + ":"
                 + dirs.get("app_packages").getAbsolutePath() + ":" + dirs.get("app").getAbsolutePath();
+        Log.d(APP_LAUNCH_TAG, "Python.init() start");
         if (Python.init(pythonHome, sysPath, null) != 0) {
             throw new Exception("Unable to start Python interpreter.");
         }
+        Log.d(APP_LAUNCH_TAG, "Python.init() complete");
 
         // Run the app's main module, similar to `python -m`.
+        Log.d(APP_LAUNCH_TAG, "Python.run() start");
         Python.run("{{cookiecutter.module_name}}", new String[0]);
+        Log.d(APP_LAUNCH_TAG, "Python.run() end");
+        Log.d(APP_LAUNCH_TAG, "startPython() end");
     }
 
     protected void onCreate(@Nullable Bundle savedInstanceState) {
+        Log.d(APP_LAUNCH_TAG, "onCreate() start");
         super.onCreate(savedInstanceState);
         this.captureStdoutStderr();
+        Log.d("MainActivity.onCreate", "captured stdout/stderr");
```

Comment:
Is this deliberately *not* APP_LAUNCH_TAG?

Reply:
I think APP_LAUNCH_TAG would be a better approach, yeah. Good catch.